luigi/contrib/dataproc.py

```diff
@@ -0,0 +1,208 @@
+"""luigi bindings for Google Dataproc on Google Cloud"""
+
+import os
+import time
+import logging
+import luigi
+
+logger = logging.getLogger('luigi-interface')
+
+_dataproc_client = None
+
+try:
+    import httplib2
+    import oauth2client.client
+    from googleapiclient import discovery
+    from googleapiclient.errors import HttpError
+
+    DEFAULT_CREDENTIALS = oauth2client.client.GoogleCredentials.get_application_default()
+    _dataproc_client = discovery.build('dataproc', 'v1', credentials=DEFAULT_CREDENTIALS, http=httplib2.Http())
+except ImportError:
+    logger.warning("Loading Dataproc module without the python packages googleapiclient & oauth2client. \
+        This will crash at runtime if Dataproc functionality is used.")
+
+
+def get_dataproc_client():
+    return _dataproc_client
+
+
+def set_dataproc_client(client):
+    global _dataproc_client
+    _dataproc_client = client
+
+
+class _DataprocBaseTask(luigi.Task):
+    gcloud_project_id = luigi.Parameter(significant=False)
+    dataproc_cluster_name = luigi.Parameter(significant=False)
+    dataproc_region = luigi.Parameter(default="global", significant=False)
+
+    dataproc_client = get_dataproc_client()
+
+
+class DataprocTask(_DataprocBaseTask):
+    """ Base task for running jobs in Dataproc """
```

Comment:
The docs should at least say that "Currently XYZ are not implemented, but we believe here is the natural place for it.". Because anyone browsing the docs online would probably be confused.